dashboard/utils.py

```diff
@@ -167,6 +174,32 @@ def has_verified_cert(self, course_id):
         """
         return self.certificates.has_verified_cert(course_id)
 
+    def extract_final_grade(self, course_id):
+        """
+        Function that extract a final grade tuple from a global class dictionary
+        or extracts the value from the database and updates such dictionary.
+
+        Args:
+            course_id (str): an edX course run id
+        Returns:
+            tuple: a tuple consisting in final_grade.grade, final_grade.passed
+        """
+        grade = self.final_grades.get(course_id)
+        if grade is not None:
+            return grade
+        try:
+            final_grade = FinalGrade.objects.get(
+                course_run__edx_course_key=course_id,
+                user=self.user,
+                status=FinalGradeStatus.COMPLETE,
+            )
+        except FinalGrade.DoesNotExist:
+            log.exception('No final grade found for user %s in course %s', self.user.username, course_id)
+            raise
+        grade = MMTrackGrade(grade=final_grade.grade, passed=final_grade.passed)
```

Comment:
The reason why I use a name tuple and not a `FinalGrade` object directly is because I cache the object and I thought it is not a good idea to cache the object from the database.